replica/database.cc

```diff
@@ -1696,6 +1696,13 @@ bool database::is_user_semaphore(const reader_concurrency_semaphore& semaphore)
         && &semaphore != &_system_read_concurrency_sem;
 }
 
+future<> database::clear_inactive_reads_for_tablet(table_id table, dht::token_range tablet_range) {
+    const auto partition_range = dht::to_partition_range(tablet_range);
+    for (auto* sem : {&_read_concurrency_sem, &_streaming_concurrency_sem, &_compaction_concurrency_sem, &_system_read_concurrency_sem}) {
+        co_await sem->evict_inactive_reads_for_table(table, &partition_range);
+    }
```

Comment:
maybe we should introduce a database::foreach_reader_concurrency_semaphore(). so integration with semaphore group becomes simpler (functions like this one and detach_column_family() remain the same regardless of the existence of the group).

Reply:
Good idea, this will reduce friction between open-source and enterprise.

Reply:
Done in v3.